staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -70,10 +70,9 @@ func ConvertJSONSchemaPropsWithPostProcess(in *apiextensions.JSONSchemaProps, ou
 	out.Description = in.Description
 	if in.Type != "" {
 		out.Type = spec.StringOrArray([]string{in.Type})
-	}
-	if in.Nullable {
-		// by validation, in.Type is either "object" or "array"
-		out.Type = append(out.Type, "null")
+		if in.Nullable {
+			out.Type = append(out.Type, "null")
```

Comment:
we only append `null` if type is not ""? does validation enforce type being non-empty?

Reply:
or does type:"" imply arbitrary type, including null?

Reply:
yes, empty type means anything. This includes the null case. So nullable is not needed.